### DISCUSSION TITLE
How are slugs defined exactly?

### DISCUSSION THREAD

**Discussion by eguiraud** [artempyanykh/marksman#210] [General]
Hello,

what's the exact transformation from note filename or title to the slug that can be used to link to it in a `[[link]]`?

Sorry if it's written somewhere, could not find it.

For example if I have a) a file called `my_note.md` with content `# My Note`, what's the correct slug? and what if b) a file called `my_note.md` does not have a `# Title` but instead only a bunch of `## Sections` ?

**Comment by artempyanykh** [comment]
Hi @eguiraud!

When it comes to filenames these are usually inserted as-is (modulo url-encoding when you have spaces in file names and special symbols). The titles are transformed using [this code](https://github.com/artempyanykh/marksman/blob/main/Marksman/Misc.fs#L77-L103): the gist is that all alphanumeric chars are lowercased and punctuation/whitespace chars are replaced with a dash '-', e.g. `"Hello world" -> "hello-world"`.

I don't think there's an official definition of what a GitHub's slug is. What I have seems like a good approximation though.

Regarding the second question:
>  b) a file called my_note.md does not have a # Title but instead only a bunch of ## Sections ?

You can still link to your file using filename `my_note` and then link to your sections in the usual way `[[my_note#My Section]]`

Hope this helps!
